python/graphstorm/config/argument.py

```diff
@@ -305,6 +310,52 @@ def set_task_attributes(self, configuration):
         for key, val in configuration.items():
             setattr(self, f"_{key}", val)
 
+    def _save_combined_config(self, output_path: str):
+        """Save a YAML file that combines the input YAML with runtime args.
+
+        Parameters:
+        -----------
+        output_path : str
+            Path where to save the config.
+        """
+        # Get the original YAML content
+        yaml_config = self.load_yaml_config(self.yaml_paths)
+
+        # Update with runtime args (all attributes starting with '_')
+        for attr_name, attr_value in vars(self).items():
+            if attr_name.startswith('_') and not attr_name.startswith('__'):
+                # Extract the key without the underscore
+                key = attr_name[1:]
+
+                # Find the appropriate section to update
+                if 'gsf' in yaml_config:
+                    for section in yaml_config['gsf'].values():
```

Comment:
What if we originally have a parameter not under gsf but override in the command? Like

```
gnn:
   lr: 0.05
```

What will the final result be here? Just want to avoid the `lr` exist in two sections, like:
```
gsf:
  lr: 0.01
gnn:
  lr: 0.05
```